File: TEST data for large repo/REPO_Copy-6/Blueprism/BluePrism.DigitalWorker/EnvironmentFunctions/GetUserNameFunction.cs
using System;
using System.Collections.Generic;
using BluePrism.AutomateAppCore;
using BluePrism.AutomateProcessCore;
using BluePrism.DigitalWorker.Sessions;

namespace BluePrism.DigitalWorker.EnvironmentFunctions
{
    public class GetUserNameFunction : GetUserNameFunctionBase
    {
        private readonly IRunningSessionRegistry _runningSessionRegistry;

        public GetUserNameFunction(IRunningSessionRegistry runningSessionRegistry)
        {
            _runningSessionRegistry = runningSessionRegistry ?? throw new ArgumentNullException(nameof(runningSessionRegistry));
        }

        protected override clsProcessValue InnerEvaluate(IList<clsProcessValue> parameters, clsProcess process)
        {
            if (process.Session == null)
                return string.Empty;

            var sessionContext = _runningSessionRegistry.Get(process.Session.ID);

            if (sessionContext == null)
                return string.Empty;

            return sessionContext.StartedByUsername;
        }
    }
}

File: TEST data for large repo/repo_Copy-2/Blueprism/BluePrism.Automate/AutomateAppCore/EnvironmentFunctions/GetUserNameFunction.vb
Imports BluePrism.AutomateProcessCore

Namespace EnvironmentFunctions

    Public Class GetUserNameFunction : Inherits GetUserNameFunctionBase
        
        Private ReadOnly mServerFactory As Func(Of IServer)

        Public Sub New(serverFactory As Func(Of IServer))
            mServerFactory = serverFactory
        End Sub

        Protected Overrides Function InnerEvaluate(parameters As IList(Of clsProcessValue), process As clsProcess) As clsProcessValue
            If parameters.Count <> 0 Then
                Throw New clsFunctionException(
                 My.Resources.GetUserNameFunction_GetUserNameFunctionShouldNotHaveAnyParameters)
            End If

            'Get the username, if there's a session to get it from
            Dim user As String = ""
            If process.Session IsNot Nothing Then
                user = mServerFactory().GetSessionDetails(process.Session.ID).UserName
            End If

            Return New clsProcessValue(DataType.text, user)

        End Function      
    End Class
End NameSpace